### DISCUSSION TITLE
Could we have "magic links" add as an auth method?

### DISCUSSION THREAD

**Discussion by richardltc** [pocketbase/pocketbase#194] [Q&A] [answered]
Could we have "magic links" add as an auth method?

Richard

**Comment by ganigeorgiev** [comment, marked as answer]
@richardltc Hi, this is already in the roadmap, but it's a low priority for now (you can subscribe to this issue - https://github.com/pocketbase/pocketbase/issues/119).

**Reply by richardltc** [reply]
Perfect. Thanks for you speedy response.  :)

**Comment by pcanterini** [comment]
So the OTP auth method requires an email record to already be present in the DB.
A magic link should allow a new user to also log in even if not already registered right?

Currently `requestOTP` only sends a code to an existing email, as far as I can tell.
Does this mean I should first create a "dummy" user with the email and a random password before sending the OTP code or is there a more elegant way of handling this use case?

**Reply by ganigeorgiev** [reply]
@pcanterini You are commenting to a very old discussion and the method you are referring to is only in the v0.23.0-rcs. Whether _"A magic link should allow a new user to also log in even if not already registered right?"_ is debatable and difficult to generalize but currently as you've noted PocketBase requires the user to exist (similar to the `requestPasswordReset` and `requestVerification` calls). Option to allow creating the user via hook was requested recently by another user and remains in my todo to think a little more on it before the final v0.23.0 release.
